src/System.Memory/src/System/Buffers/OwnedMemory.cs

```diff
@@ -0,0 +1,97 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime;
+using System.Runtime.CompilerServices;
+
+namespace System.Buffers
+{
+
+    /// <summary>
+    /// Owner of Memory<typeparamref name="T"/> that provides appropriate lifetime management mechanisms for it.
+    /// </summary>
+    public abstract class OwnedMemory<T> : IDisposable, IRetainable
+    {
+
+        /// <summary>
+        /// The number of items in the Memory<typeparamref name="T"/>.
+        /// </summary>
+        public abstract int Length { get; }
+
+        /// <summary>
+        /// Returns a span wrapping the underlying memory.
+        /// </summary>
+        public abstract Span<T> AsSpan();
+
+        /// <summary>
+        /// Returns a Memory<typeparamref name="T"/> if the underlying memory has not been freed.
+        /// </summary>
+        /// <exception cref="System.ObjectDisposedException">
+        /// Throw when the memory has been disposed.
```

Comment:
this is a bit misleading. It implies that the dispose would need to be called on this Memory<T>, where in fact somebody else could call Dispose on the OwnedMemory.

Reply:
I am not really sure what you mean. I didn't specify who called Dispose. I changed it to "Thrown when the underlying memory has already been disposed."